### PR TITLE
KHWE-24: Implement playing song from a list

### PR DESCRIPTION
**Pull request by lubovgribiniyk**

### CODE REVIEW

File: src/components/MySongsTable/MySongsTable.js [hunk 1]
```diff
@@ -68,13 +56,17 @@ export class MySongsTableComponent {
   }
 
   mount() {
-    this.mountPoint.innerHTML = this.render();
-    this.querySelectors();
-    this.initMaterial();
-    this.mountChildren();
+    MusicService.getAlbumSongs("album1").then(songs => {
```
Comment (ihor4568) [inline src/components/MySongsTable/MySongsTable.js]: no. First you should render empty table. And then fetch data. And then render table with data

File: src/components/MediaPlayer/MediaPlayer.js [hunk 2]
```diff
@@ -76,6 +77,25 @@ export class MediaPlayerComponent {
     window.open("/song/awdklawj");
   }
 
+  setNewSong(song) {
+    const albumPromise = MusicService.getAlbumById(song.albumId);
```
Comment (ihor4568) [inline src/components/MediaPlayer/MediaPlayer.js]: the data about artist and album should be passed as parameters of this method. You probably already have them in songs table

File: src/components/Main/Main.js [hunk 3]
```diff
@@ -128,7 +132,9 @@ export class MainComponent {
     this.shareView = new ShareViewComponent(this.shareViewPoint);
 
     this.about = new AboutComponent(this.mainPoint);
-    this.table = new MySongsTableComponent(this.mainPoint);
+    this.table = new MySongsTableComponent(this.mainPoint, {
+      handlePlaySong: this.handlePlaySong.bind(this)
```
Comment (ihor4568) [inline src/components/Main/Main.js]: the property should be called `onSongPlay`. and handler name should be `handleSongPlay`

File: src/components/MySongsTable/MySongsTable.js (hunk 1 above)
Comment (lubovgribiniyk) [reply src/components/MySongsTable/MySongsTable.js]: If I correctly understand the task is just to come up with approach for handling playing song from a list. The pages with these tables are developed by other guys (KHWE-22, KHWE-23). Does it make sense to fix this logic here?

File: src/components/MediaPlayer/MediaPlayer.js (hunk 2 above)
Comment (lubovgribiniyk) [reply src/components/MediaPlayer/MediaPlayer.js]: Table components (ArtistSongsTable or AlbumSongsTable) are no ready yet, so I am not sure how these data will be implemented in these components.

File: src/components/MySongsTable/MySongsTable.js (hunk 1 above)
Comment (ihor4568) [reply src/components/MySongsTable/MySongsTable.js]: yes. Your task is to implement playing from my songs list.

File: src/components/MediaPlayer/MediaPlayer.js (hunk 2 above)
Comment (ihor4568) [reply src/components/MediaPlayer/MediaPlayer.js]: implement ONLY for mysongs table

File: src/components/MySongs/MySongs.js [hunk 4]
```diff
@@ -39,14 +15,43 @@ export class MySongsComponent {
...
+      Promise.all(
+        this.songs.map(song => {
+          const albumPromise = MusicService.getAlbumById(song.albumId);
+          const authorsPromise = Promise.all(
+            song.authors.map(authorId => MusicService.getAuthorById(authorId))
+          );
+
+          const songInfoPromises = Promise.all([albumPromise, authorsPromise]);
+          songInfoPromises.then(([album, authors]) => {
+            /* eslint-disable no-param-reassign */
```
Comment (ihor4568) [inline src/components/MySongs/MySongs.js]: please do not disable this rule

File: src/components/MySongs/MySongs.js [hunk 5]
```diff
@@ -39,14 +15,43 @@ export class MySongsComponent {
     );
   }
 
+  fetchSongs() {
+    MusicService.getAlbumSongs("album1").then(songs => {
```
Comment (ihor4568) [inline src/components/MySongs/MySongs.js]: please fix callback hell here

File: src/components/SongsTable/SongsTable.js [hunk 6]
```diff
@@ -63,6 +63,15 @@ export class SongsTableComponent {
     this.mount();
   }
 
+  handlePlayClick(e) {
+    const target = e.target.closest(".songs-table__td_play-btn");
```
Comment (ihor4568) [inline src/components/SongsTable/SongsTable.js:67]: actually you should check here if target is a button that has been supposed to be clicked.

Comment (lubovgribiniyk) [reply src/components/SongsTable/SongsTable.js:67]: I find the button by selector here and check it below. Potentially the click can be on the `span` inside button so I use `closest` to make sure that the button was clicked.

**Review by ihor4568 — APPROVED** [review]

**Pull request merged**